src/mmds/src/data_store.rs

```diff
@@ -107,39 +99,7 @@ impl Mmds {
         };
 
         match value {
-            Some(val) => {
-                let mut ret = Vec::new();
-                // If the `dict` is Value::Null, Error::NotFound is thrown.
-                // If the `dict` is not a dictionary, a Vec with the value corresponding to
-                // the key is returned.
-                match val.as_object() {
-                    Some(map) => {
-                        // When the object is a map, push all the keys in the Vec.
-                        for key in map.keys() {
-                            let mut key = key.clone();
-                            // If the key corresponds to a dictionary, a "/" is appended
-                            // to the key name.
-                            if map[&key].is_object() {
-                                key.push_str("/");
-                            }
-
-                            ret.push(key);
-                        }
-                        Ok(ret)
-                    }
-                    None => {
-                        // When the object is not a map, return the value.
-                        // The only supported Value type is String.
-                        match val.as_str() {
-                            Some(str_val) => {
-                                ret.push(str_val.to_string());
-                                Ok(ret)
-                            }
-                            None => Err(Error::UnsupportedValueType),
-                        }
-                    }
-                }
-            }
+            Some(json) => Ok(json.to_string()),
```

Comment:
This can result in invalid JSON values. For example, in the tests where you do `mmds.get_value("/phones2".to_string()).unwrap()`, the return is something that is not a json.

This is breaking the contract of the function which specifies that it returns a json. There are a few options to fix this, but I am not sure which one is best.

1. Check that the path actually corresponds to a json (i.e. you cannot do get on a key that has as value a string/float/vec). This seems very restrictive.
2. When the value of a key is not JSON, create a valid JSON from it by doing something like:
``'{' + key + ':' + value '}'` (this is pseudo code :)) )

These are the only two I can think of, but there are probably others as well.

Reply:
I would go with second option. Also I would add a unit test for this corner case that would validate with a tool that the json is in proper format.

Reply:
Disregard my comment. I read through the RFC again and apparently strings are valid jsons :))

https://tools.ietf.org/html/rfc7159#section-3